### DISCUSSION TITLE
gdb cannot load symbols

### DISCUSSION THREAD

**Discussion by GhislainViguier** [nlohmann/json#4128] [Q&A]
I'm having an issue with json and gdb.
In a simple C++ executable, I load a json file with nlohmann/json :
```
auto cmd = json::parse(std::ifstream(jsonFile));
```
It works well but when I try to debug with gdb on my Linux, gdb cannot load debug symbols. I get :
```
Reading symbols from myExe ...Unexpected type (10) encountered for integer constant.
(no debugging symbols found)...done.
```

If I remove all the json calls, there is no issue, gdb can load symbols.
The code was compiled with intel :
```
$ icpc --version
icpc (ICC) 2021.5.0 20211109
```
And my system is Red Hat Linux Server release 7.6 (Maipo)

Have you encountered this issue? Do you know how I can fix it?

**Comment by nlohmann** [comment]
This looks unrelated to nlohmann/json's own code — 'Unexpected type (10) encountered for integer constant' is GDB failing to parse DWARF debug info emitted by the Intel C++ Compiler (icpc), a known incompatibility between older/newer DWARF versions produced by icc and the GDB version shipped with RHEL 7.6. Since removing the json.hpp calls avoids it, the header is likely just pulling in enough template instantiations/constexpr constants to trigger a DWARF construct icc emits that this GDB can't parse. Try compiling with -gdwarf-2 (or whatever max DWARF version your GDB supports) via icpc's debug flags, or upgrade GDB; this isn't something the JSON library can control.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
